stan/math/rev/functor/adj_jac_apply.hpp

```diff
@@ -99,19 +96,17 @@ class adj_jac_vari : public vari {
   /**
    * Return an eigen matrix of the adjoints
    * @tparam RetType Not called by user. Default of the return type of
-   * operator() of the functor F allows for SFINAE to choose the correct
+   * `forward_pass()` of the functor F allows for SFINAE to choose the correct
    * specialization given the required output type.
    */
-  template <typename RetType = FReturnType, require_eigen_t<RetType>* = nullptr>
+  template <typename RetType = FunctorReturnType,
+            require_eigen_t<RetType>* = nullptr>
   inline auto y_adj() const {
-    Eigen::Matrix<double, FReturnType::RowsAtCompileTime,
-                  FReturnType::ColsAtCompileTime>
-        var_y(dims_[0], dims_[1]);
-    const size_t iter_size{dims_[0] * dims_[1]};
-    for (size_t i = 0; i < iter_size; ++i) {
-      var_y(i) = y_vi_[i]->adj_;
-    }
-    return var_y;
+    return Eigen::Map<Eigen::Matrix<vari*, FunctorReturnType::RowsAtCompileTime,
+                                    FunctorReturnType::ColsAtCompileTime>>(
```

Comment:
Actually we can further simplify that.
```suggestion
    return Eigen::Map<promote_scalar_t<vari*, FunctorReturnType>>(
```